src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java

```diff
@@ -133,14 +132,10 @@ private static boolean isEqualsMethod(DetailAST ast) {
      */
     private static boolean isHashCodeMethod(DetailAST ast) {
         final DetailAST modifiers = ast.getFirstChild();
-        final DetailAST type = ast.findFirstToken(TokenTypes.TYPE);
         final DetailAST methodName = ast.findFirstToken(TokenTypes.IDENT);
         final DetailAST parameters = ast.findFirstToken(TokenTypes.PARAMETERS);
 
-        return type.getFirstChild().getType() == TokenTypes.LITERAL_INT
```

Comment:
> public Integer hashCode()

cannot override hashCode() in java.lang.Object return type java.lang.Integer is not compatible with int